src/nba/nbaPlayerProjection.ts

```diff
@@ -0,0 +1,27 @@
+import { PlayerProjection } from "../types/PlayerProjection.js";
+import { NBAStatSet } from "./nbaStatSet.js";
+import { FantasyStatValueMap } from "../types/FantasyStats.js";
+import { StatProductionType } from "../types/FantasyStats.js";
+
+export class NBAPlayerProjection implements PlayerProjection {
+    fantasyPoints: number;
+
+    constructor(statProjection: NBAStatSet, statValueMap: FantasyStatValueMap) {
+
+        if (statProjection.productionType === StatProductionType.ACTUAL) {
+            throw new Error('Player projection must be based on projected stats, not actual production.')
+        }
+
+        this.fantasyPoints = this.getProjectedFantasyPoints(statProjection, statValueMap);
+    }
+
+    public getProjectedFantasyPoints(statProjection: NBAStatSet, statValueMap: FantasyStatValueMap): number {
```

Comment:
can be private, unless you intend for users of the class to call this and change the fantasyPoints, in which case, this might be misnamed